src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -38,30 +38,24 @@ using namespace Start;
 namespace
 {
 
-std::string humanReadableSize(unsigned int bytes)
+static std::string humanReadableSize(const size_t size)
 {
-    static const std::vector<std::string> siPrefix {
-        "b",
-        "kb",
-        "Mb",
-        "Gb",
-        "Tb",
-        "Pb",
-        "Eb"  // I think it's safe to stop here (for the time being)...
-    };
-    size_t base = 0;
-    double inUnits = bytes;
-    constexpr double siFactor {1000.0};
-    while (inUnits > siFactor && base < siPrefix.size() - 1) {
-        ++base;
-        inUnits /= siFactor;
-    }
-    if (base == 0) {
-        // Don't include a decimal point for bytes
-        return fmt::format("{:.0f} {}", inUnits, siPrefix[base]);
-    }
-    // For all others, include one digit after the decimal place
-    return fmt::format("{:.1f} {}", inUnits, siPrefix[base]);
+    static constexpr std::array<std::pair<char, unsigned long long>, 7> prefixes {
+        {{0, 0},
+         {'k', 1ULL << 10},
+         {'M', 1ULL << 20},
+         {'G', 1ULL << 30},
+         {'T', 1ULL << 40},
+         {'P', 1ULL << 50},
+         {'E', 1ULL << 60}}};

```

Comment:
I guess that it is fine, but honestly I'm not sold at the bitwise shift here. It needs a moment to understand. I'm also not sold on the char concept, being explicit here and providing full unit i.e. kB would be much more readable. It will also not leave this akward 0.

Reply:
Consider I was even temped to use integer math instead of typecasting to `double` ;-). This is just 2¹⁰, 2²⁰ (roughly 10³, 10⁶)... expressed in usual C way. Each value could be made `constexpr` with some nice descriptive name providing meaning. But is not that already provided by prefix right before it?

Reply:
I know what it is, but I it is again _clever_ code. In order to understand it you need to know that 1024 is 2^10 and so each magnitude transfers nicely to nice round powers. A bit of comment would help with that though. Something like 
```
// 1ULL << 10 = 2^10 = 1024
// 1ULL << 20 = 2^20 = 1024 * 1024
// etc
```

You need to know that left bit shift is essentially taking 2 to the power. I know that stuff, you need that stuff but I need to take a moment to ensure that it indeed is what I see. 

https://read.engineerscodex.com/p/clever-code-is-probably-the-worst

Code should be clear, easy to parse - you should not waste brain cycles to understand what is going on.